R/data-combined.R

```diff
@@ -0,0 +1,807 @@
+      private$.groupMap <- private$.extractGroupMap(private$.dataCombinedDF)
+
+      # update dataset names
+      private$.names <- private$.extractNames(private$.dataCombinedDF)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    #' @description
+    #'
+    #' Transform raw data with required offsets and scale factors.
+    #'
+    #' @param xOffsets,yOffsets,xScaleFactors,yScaleFactors Either a numeric
+    #'   scalar or a list of numeric quantities specifying offsets and scale
+    #'   factors to apply to raw values. The default offset is `0`, while
+    #'   default scale factor is `1`, i.e., the data will not be modified. If a
+    #'   list is specified, it should be the same length as `names` argument.
+    #'
+    #' @return A dataframe with respective raw quantities plus offsets
+    #'   multiplied by the specified scale factors. If error column is present,
+    #'   it will also be scaled.
+
+    setDataTransforms = function(names = NULL,
+                                 xOffsets = 0,
+                                 yOffsets = 0,
+                                 xScaleFactors = 1,
+                                 yScaleFactors = 1) {
+      # check that the arguments to parameters make sense
+      validateIsString(names, nullAllowed = TRUE)
+      validateIsNumeric(xOffsets)
+      validateIsNumeric(yOffsets)
+      validateIsNumeric(xScaleFactors)
+      validateIsNumeric(yScaleFactors)
+
+      # transform data
+      if (!is.null(private$.dataCombinedDF)) {
+        private$.dataCombinedDF <- private$.dataTransform(
+          data          = private$.dataCombinedDF,
+          names         = names,
+          xOffsets      = xOffsets,
+          yOffsets      = yOffsets,
+          xScaleFactors = xScaleFactors,
+          yScaleFactors = yScaleFactors
+        )
+
+        # extract a dataframe with data transformations values
+        private$.dataTransformations <- private$.extractTransforms(private$.dataCombinedDF)
+      }
+
+      # for method chaining
+      invisible(self)
+    },
+
+    ## getter methods ---------------
+
+    #' @description
+    #' A method to extract a dataframe of simulated and/or observed data
+    #' (depending on instances of which objects have been added to the object).
+    #'
+    #' Note that the order in which you enter different object doesn't matter
+    #' because the returned dataframe is arranged alphabetically by dataset
+    #' name.
+    #'
+    #' @return
+    #'
+    #' A dataframe.
+    #'
+    #' The length (number of columns) and number of rows of a dataframe will
+    #' change depending on instances of which objects and how many objects have
+    #' been provided to the object. During the object's lifetime, this method
+    #' will return an updated version of a dataframe that combines observed and
+    #' simulated data contained in provided objects.
+    #'
+    #' For the same reasons, the dataframe might contain some or all of the
```

Comment:
This is not entirely correct. The columns depend on the meta data stored in the `DataSet` objects. The number or names of meta data are not restricted (e.g. Source, Sheet, etc are meta data).

Reply:
I've updated the description with a copy-paste of what you have written here.